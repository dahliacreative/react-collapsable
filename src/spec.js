import React from 'react'
import { mount } from 'enzyme'
import Collapsable from '.'

const defineScrollHeight = height => {
    Object.defineProperty(HTMLElement.prototype, 'scrollHeight', {
        configurable: true,
        get: function() {
            return height
        }
    })
}

defineScrollHeight(500)

describe('<Collapsable/>', () => {
    it('Renders without crashing', () => {
        const component = mount(
            <Collapsable>
                <h1>My Title</h1>
            </Collapsable>
        )
        expect(component).toMatchSnapshot()
    })

    it('Renders open by default', () => {
        const component = mount(
            <Collapsable isOpen>
                <h1>My Title</h1>
            </Collapsable>
        )
        expect(component).toMatchSnapshot()
    })

    it('Opens', done => {
        const component = mount(
            <Collapsable>
                <h1>My Title</h1>
            </Collapsable>
        )
        component.setProps({ isOpen: true })
        setTimeout(() => {
            expect(component).toMatchSnapshot()
            done()
        }, 500)
    })

    it('Closes', done => {
        const component = mount(
            <Collapsable isOpen>
                <h1>My Title</h1>
            </Collapsable>
        )
        component.setProps({ isOpen: false })
        setTimeout(() => {
            expect(component).toMatchSnapshot()
            done()
        }, 500)
    })

    it('Respects maxAnimationDuration', done => {
        defineScrollHeight(2000)
        const component = mount(
            <Collapsable>
                <h1>My Title</h1>
            </Collapsable>
        )
        component.setProps({ isOpen: true })
        setTimeout(() => {
            expect(component).toMatchSnapshot()
            done()
        }, 1000)
    })

    it('Respects minAnimationDuration', done => {
        defineScrollHeight(100)
        const component = mount(
            <Collapsable>
                <h1>My Title</h1>
            </Collapsable>
        )
        component.setProps({ isOpen: true })
        setTimeout(() => {
            expect(component).toMatchSnapshot()
            done()
        }, 300)
    })
})
