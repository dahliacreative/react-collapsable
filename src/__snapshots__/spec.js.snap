// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Collapsable/> Closes 1`] = `
<Collapsable
  easing="ease-in-out"
  isOpen={false}
  maxAnimationDuration={1}
  minAnimationDuration={0.3}
  speedDivider={1000}
>
  <div
    onTransitionEnd={[Function]}
    style={
      Object {
        "height": 500,
        "overflow": "hidden",
        "transition": "height 0.5s ease-in-out",
      }
    }
  >
    <div
      style={
        Object {
          "overflow": "auto",
        }
      }
    >
      <h1>
        My Title
      </h1>
    </div>
  </div>
</Collapsable>
`;

exports[`<Collapsable/> Opens 1`] = `
<Collapsable
  easing="ease-in-out"
  isOpen={true}
  maxAnimationDuration={1}
  minAnimationDuration={0.3}
  speedDivider={1000}
>
  <div
    onTransitionEnd={[Function]}
    style={
      Object {
        "height": 0,
        "overflow": "hidden",
        "transition": "height 0s ease-in-out",
      }
    }
  >
    <div
      style={
        Object {
          "overflow": "auto",
        }
      }
    />
  </div>
</Collapsable>
`;

exports[`<Collapsable/> Renders open by default 1`] = `
<Collapsable
  easing="ease-in-out"
  isOpen={true}
  maxAnimationDuration={1}
  minAnimationDuration={0.3}
  speedDivider={1000}
>
  <div
    onTransitionEnd={[Function]}
    style={
      Object {
        "height": 500,
        "overflow": "hidden",
        "transition": "height 0.5s ease-in-out",
      }
    }
  >
    <div
      style={
        Object {
          "overflow": "auto",
        }
      }
    >
      <h1>
        My Title
      </h1>
    </div>
  </div>
</Collapsable>
`;

exports[`<Collapsable/> Renders without crashing 1`] = `
<Collapsable
  easing="ease-in-out"
  isOpen={false}
  maxAnimationDuration={1}
  minAnimationDuration={0.3}
  speedDivider={1000}
>
  <div
    onTransitionEnd={[Function]}
    style={
      Object {
        "height": 0,
        "overflow": "hidden",
        "transition": "height 0s ease-in-out",
      }
    }
  >
    <div
      style={
        Object {
          "overflow": "auto",
        }
      }
    />
  </div>
</Collapsable>
`;

exports[`<Collapsable/> Respects maxAnimationDuration 1`] = `
<Collapsable
  easing="ease-in-out"
  isOpen={true}
  maxAnimationDuration={1}
  minAnimationDuration={0.3}
  speedDivider={1000}
>
  <div
    onTransitionEnd={[Function]}
    style={
      Object {
        "height": 0,
        "overflow": "hidden",
        "transition": "height 0s ease-in-out",
      }
    }
  >
    <div
      style={
        Object {
          "overflow": "auto",
        }
      }
    />
  </div>
</Collapsable>
`;

exports[`<Collapsable/> Respects minAnimationDuration 1`] = `
<Collapsable
  easing="ease-in-out"
  isOpen={true}
  maxAnimationDuration={1}
  minAnimationDuration={0.3}
  speedDivider={1000}
>
  <div
    onTransitionEnd={[Function]}
    style={
      Object {
        "height": 0,
        "overflow": "hidden",
        "transition": "height 0s ease-in-out",
      }
    }
  >
    <div
      style={
        Object {
          "overflow": "auto",
        }
      }
    />
  </div>
</Collapsable>
`;
